tools/topology/sof/pipe-detect.m4

```diff
@@ -0,0 +1,76 @@
+# Sound Detector
+#
+#  Generic sound detector.
+#
+# Pipeline Endpoints for connection are :-
+#
+#  (Sound Detector <-- Detect Switch) <-- Channel Mux <--- Source Pipeline
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`buffer.m4')
+include(`pga.m4')
+include(`detect.m4')
+include(`mixercontrol.m4')
+include(`enumcontrol.m4')
+include(`pipeline.m4')
+
+#
+# Controls
+#
+
+# Switch type Mixer Control with max value of 1 (Internal to Detector)
+C_CONTROLENUM_TEXT(CHANNELS, LIST(`	        ', "Left", "Right"))
+C_CONTROLMIXER(Detect Switch, PIPELINE_ID,
+	CONTROLMIXER_OPS(volsw, 259 binds the mixer control to switch get/put handlers, 259, 259),
+	CONTROLMIXER_MAX(max 1 indicates switch type control, 1),
+	false,
+	,
+	Channel register and shift for Front Left/Right,
+	LIST(`	', KCONTROL_CHANNEL(FL, 2, 0), KCONTROL_CHANNEL(FR, 2, 1)))
+
+# Channel Selector Mux control with max value of 32 (Internal to Detector)
+# TODO pass in enum options
+C_CONTROLENUM(Channel Mux, PIPELINE_ID,
+	CONTROLENUM_OPS(enum, 257 binds the mixer control to enum get/put handlers, 257, 257),
+	LIST(`	', KCONTROL_ENUM_CHANNEL(FL, 0, 0), KCONTROL_ENUM_CHANNEL(FR, 0, 1)),
+	CHANNELS)
+
+#
+# Components and Buffers
+#
+
+# "Detect 0" has 2 sink period and 0 source periods
+W_DETECT(0, PIPELINE_FORMAT, 2, 0, KEYWORD,
+	LIST(`		', "Detect Switch PIPELINE_ID"))
+
+W_MUX(0, PIPELINE_FORMAT, 2, 2, SOF_COMP_MUX_CH_SEL, LIST(`		', "Channel Mux PIPELINE_ID"), PIPELINE_ID)
```

Comment:
What values of "LIST(`		', "Channel Mux PIPELINE_ID")" will be provided in the topology  0,1 or "Left, Right" ? how they will be maped in sof_ipc_ctrl_cmd structure ? Assuming cmd == SOF_CTRL_CMD_ENUM  I suppose we need to add a value field in the union in sof_ipc_ctrl_cmd.

Reply:
@lbetlej topology will provide "left" and "Right" and these are what will show up with amixer cget.
But what the driver will see is a  0 or 1 depending on whats defined as item 0 and item 1 in topology for "left" and "right". And the fw will eventually get a 0/1 based on the user selection. does this make sense?